webapp/src/js/controllers/messages-content.js

```diff
@@ -80,37 +81,32 @@ angular.module('inboxControllers').controller('MessagesContentCtrl',
       $('.message-content-wrapper').on('scroll', checkScroll);
     };
 
-    var markAllRead = function() {
-      var docs = _.pluck(ctrl.selectedMessage.messages, 'doc');
-      if (docs.length) {
-        $scope.markConversationRead(docs);
-        MarkRead(docs)
-          .catch(function(err) {
-            return $log.error('Error marking all as read', err);
-          });
-      }
-    };
-
     // See $stateParams.id note at top of file
     var getContactable = function(id, type) {
       if (type === 'contact') {
-        return LineageModelGenerator.contact(id)
-                                    .catch(err => {
-                                      if (err.code === 404) {
-                                        return Promise.resolve();
-                                      }
-
-                                      throw err;
-                                    });
+        return LineageModelGenerator.contact(id).catch(err => {
+          if (err.code === 404) {
+            return Promise.resolve();
+          }
+          throw err;
+        });
       } else if (type === 'phone') {
         return {name: id};
       } else {
         return {};
       }
     };
 
-    var selectContact = function(id, type, options) {
-      options = options || {};
+    const markConversationReadIfNeeded = () => {
+      const hasUnreadDoc = ctrl.selectedMessage.messages.some(message => {
+        return !message.read && message.doc;
+      });
+      if (hasUnreadDoc) {
+        ctrl.markConversationRead();
+      }
+    };
```

Comment:
This makes the e2e test more stable (as well as the actual performance better) by only updating the redux state (and forcing an angular render) when the doc is not already read.